test/e2e/locators/recordset.ts

```diff
@@ -161,6 +165,68 @@ export default class RecordsetLocators {
     return container.locator('.side-panel-resizable');
   }
 
+  static getFacetById(container: Page | Locator, idx: number): Locator {
+    return container.locator(`.fc-${idx}`);
+  }
+
+  static getFacetHeaderButtonById(facet: Locator, idx: number): Locator {
+    return facet.locator(`.fc-heading-${idx} button`);
+  }
+
+  // get child of accordion group, sibling to accordion heading
+  static getFacetCollapse(facet: Locator): Locator {
+    return facet.locator('.accordion-collapse');
+  }
+
+  static getFacetOptions(facet: Locator): Locator {
+    return facet.locator('.chaise-checkbox label')
+  }
+
+  static getFacetOption(facet: Locator, optionIdx: number) {
+    return facet.locator(`.checkbox-${optionIdx}`);
+}
 
+  /* range facet selectors */
+  // there's integer/float/date/timestamp inputs
+  static getRangeInput(facet: Locator, className: string): Locator {
+    return facet.locator(`.${className}`);
+  }
+
+  static getInputClear(facet: Locator, className: string): Locator {
+    // NOTE: same as getRangeInput selector
+    return facet.locator(`.${className}`);
+  }
+
+  static getRangeSubmit(facet: Locator): Locator {
+    return facet.locator('.range-input-submit-btn');
+  }
+
+  /* histogram selectors */
+  static getHistogram(facet: Locator): Locator {
+    return facet.locator('.js-plotly-plot');
+  };
+
+
+  // ---------------- saved query selector ------------------- //
+
+  static getSavedQueryDropdown(container: Page | Locator): Locator {
+    return container.locator('.saved-query-menu button')
+  }
+
+  // all dropdown menu items
+  static getSavedQueryOptions(container: Page | Locator): Locator {
+    return container.locator('.saved-query-menu-item');
+  }
+
+  // recordedit option to save a query
+  static getSaveQueryOption(container: Page | Locator): Locator {
+    // substring matching
+    return container.locator('text=Save current search criteria');
```

Comment:
I didn't know you could do something like this. Playwright documentation actually encourages us to use [`getByText()`](https://playwright.dev/docs/locators#locate-by-text) which is similar to what you have done here. That way instead of relying the actual implementation details (class, id, etc), we're going by what users sees.

But given that this required a lot of rethinking in our locators, I kept using what we had. As long as we make sure we're keeping all these implementation details in the locators, we could eventually change all to use [these recommended locators](https://playwright.dev/docs/locators) instead.